app/controllers/ally_controller.rb

```diff
@@ -0,0 +1,177 @@
+#
+
+# @API Ally
+# API for access Ally data
+class AllyController < ApplicationController
+
+  # @API Get Ally information
+  # Returns the Ally integration information
+  #
+  # @example_request
+  #
+  #   curl 'https://<canvas>/api/v1/ally' \
+  #         -H 'Authorization: Bearer <token>'
+  #
+  # @example_response
+  #
+  #  { "enabled": true, "clientId": 47831, "baseUrl": "https://prod.ally.ac" }
+  #
+  def enabled
+    get_context
+    ally_settings = get_ally_settings
+    data = {
+      :enabled => ally_settings[:enabled],
+      :clientId => ally_settings[:client_id],
+      :baseUrl => ally_settings[:base_url]
+    }
+    render :json => data
+  end
+
+  # @API Create a signed request to the Ally API
+  # Get the OAuth signature so a request to the Ally API can be made
+  #
+  # @example_request
+  #
+  #   curl 'https://<canvas>/api/v1/courses/<course_id>/ally/sign?http_method=<GET|POST>&http_path=<path>&http_parameters=<url encoded parameters>' \
+  #         -H 'Authorization: Bearer <token>'
+  #
+  # @example_response
+  #
+  #  { "clientId": 47831, "baseUrl": "https://prod.ally.ac", "path": "<path>", "header": "87f82d487f78ffae4dd8b8edaa2547d7", "body": "<urlencoded body>" }
+  #
+  def sign
+    get_context
+    request = get_signed_request_object()
+    if (request)
+      ally_settings = get_ally_settings()
+      data = {
+        :clientId => ally_settings[:client_id],
+        :baseUrl => ally_settings[:base_url],
+        :path => request.path,
+        :header => request.to_hash['authorization'][0],
+        :body => request.body,
+      }
+      render :json => data
+    end
+  end
+
+  ##API Proxy a request to the Ally API
+  # Proxy data from the Ally REST API
+  #
+  # @example_request
+  #
+  #   curl 'https://<canvas>/api/v1/courses/<course_id>/ally/proxy?http_method=<GET|POST>&http_path=<path>&http_parameters=<url encoded parameters>' \
+  #         -H 'Authorization: Bearer <token>'
+  def proxy
+    get_context
+    request = get_signed_request_object()
+    if request
+      ally_settings = get_ally_settings()
+      uri = URI.parse(ally_settings[:base_url])
+      http = Net::HTTP.new(uri.host, uri.port)
+      response = http.request(request)
+      # Render successful requests as JSON
+      if response.code == 200 || response.code == 201
+        render :json => response.body, :status => response.code
+      else
+        render :text => response.body, :status => response.code
+      end
+    end
+  end
+
+  protected
+
+  ##
+  # Get a request object that can be constructed for the parameters
+  # to the `sign` or `proxy` endpoints
+  #
+  # If the current request context fails any prerequisites, an appropriate
+  # error response will be rendered and `false` will be returned
+  def get_signed_request_object
+    # Respond with a 400 if the Ally integration hasn't been enabled
+    ally_settings = get_ally_settings
+    if !ally_settings[:enabled]
```

Comment:
Looked into it a bit, in Ruby only `false` and `nil` are falsey, all other values are truthy. That said, there are only a few lines in Canvas their codebase that do `if foo == true`.

I made the change anyway as it might be clearer when they're reviewing and they can then get back to us about whether this fails there styleguide